### PR TITLE
Upgrade HBase 2.x to 2.0.0-beta-1

### PR DESCRIPTION
**Pull request by pawan-qlogic**
Upgrade HBase 2.x to 2.0.0-beta-1 and Upgraded Guava version from 19.0 to 20.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -62,7 +62,7 @@ limitations under the License.
         <protobuff-java.version>3.5.1</protobuff-java.version>
         <google.api.client.version>1.23.0</google.api.client.version>
         <google.http.client.version>1.23.0</google.http.client.version>
-        <guava.version>19.0</guava.version>
+        <guava.version>20.0</guava.version>
```
Comment (sduskis) [inline pom.xml]: Please keep guava at 19.0

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 2]
```diff
@@ -241,7 +253,6 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
     return CompletableFuture.completedFuture(!disabledTables.contains(tableName));
   }
 
-  @Override
   public CompletableFuture<TableDescriptor> getTableDescriptor(TableName tableName) {
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: If this method is removed from the HBase interface, please remove this method, and have all uses of this method use `getDescriptor`

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 3]
```diff
@@ -750,4 +608,281 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
+  }
+
+  @Override
+  public <S, R> CompletableFuture<R> coprocessorService(Function<RpcChannel, S> arg0, ServiceCaller<S, R> arg1,
+      ServerName arg2) {
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
+  @Override
+  public CompletableFuture<Void> createTable(TableDescriptor arg0) {
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: This method, and some of the ones below it should be implementable.  Pease add a comment above this method that this should be implemented.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 4]
```diff
@@ -750,4 +608,281 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
+  @Override
+  public CompletableFuture<Void> deleteSnapshots(Pattern arg0) {
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
+  @Override
+  public CompletableFuture<Void> deleteTableSnapshots(Pattern arg0) {
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: Everything until here should be implementable.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 5]
```diff
@@ -750,4 +608,281 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
+  public CompletableFuture<Boolean> isNormalizerEnabled() {
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
+  @Override
+  public CompletableFuture<Boolean> isSplitEnabled() {
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
+  @Override
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: This method should be implementable

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 6]
```diff
@@ -750,4 +608,281 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
+  public CompletableFuture<List<ReplicationPeerDescription>> listReplicationPeers(Pattern arg0) {
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
+  @Override
+  public CompletableFuture<List<SnapshotDescription>> listSnapshots() {
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: this method should be implementable

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 7]
```diff
@@ -750,4 +608,281 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
+  @Override
+  public CompletableFuture<List<SnapshotDescription>> listSnapshots(Pattern arg0) {
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: this method should be implementable

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 8]
```diff
@@ -750,4 +608,281 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
+  }
+
+  @Override
+  public CompletableFuture<List<TableDescriptor>> listTableDescriptors(boolean arg0) {
+    throw new UnsupportedOperationException("updateConfiguration"); // TODO
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: this method should be implementable

File: pom.xml (hunk 1 above)
Comment (pawan-qlogic) [reply pom.xml]: If I keep Guava 19.0 then I get below error for integration test execution -

Results :

Failed tests: 
  TestAsyncAdmin.testCreateTable_exception 
Expected: an instance of java.lang.NullPointerException
     but: <java.lang.UnsupportedOperationException: createTable> is a java.lang.UnsupportedOperationException
Stacktrace was: java.lang.UnsupportedOperationException: createTable
	at com.google.cloud.bigtable.hbase2_x.BigtableAsyncAdmin.createTable(BigtableAsyncAdmin.java:685)
	at com.google.cloud.bigtable.hbase.async.TestAsyncAdmin.testCreateTable_exception(TestAsyncAdmin.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:393)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Tests in error: 
  TestAsyncAdmin.testCreateTable_harness:86 » UnsupportedOperation createTable
  TestAsyncAdmin.testDisbleTable_exceptions:213 » UnsupportedOperation createTab...
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCheckAndDeleteDiffQual:177->checkAndDelete:81->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCheckAndDeleteDiffRow:242->checkAndDelete:81->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCheckAndDeleteSameQual:105->checkAndDelete:81->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCheckAndMutate:264->checkAndMutate:91->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCheckAndPutDiffQual:136->checkAndPut:61->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCheckAndPutDiffRow:219->checkAndPut:61->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCheckAndPutSameQual:54->checkAndPut:61->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCompareOps:313->checkAndPut:71->getBuilder:99 » NoClassDefFound
  TestAsyncCheckAndMutate>AbstractTestCheckAndMutate.testCompareOpsVersions:386->checkAndPut:71->getBuilder:99 » NoClassDefFound

Tests run: 163, Failures: 1, Errors: 11, Skipped: 1

Reason is below 

-------------------------------------------------------------------------------
Test set: com.google.cloud.bigtable.hbase.async.TestAsyncCheckAndMutate
-------------------------------------------------------------------------------
Tests run: 9, Failures: 0, Errors: 9, Skipped: 0, Time elapsed: 0.317 sec <<< FAILURE! - in com.google.cloud.bigtable.hbase.async.TestAsyncCheckAndMutate
testCheckAndDeleteDiffRow(com.google.cloud.bigtable.hbase.async.TestAsyncCheckAndMutate)  Time elapsed: 0.008 sec  <<< ERROR!
java.lang.NoClassDefFoundError: org/apache/hadoop/hbase/shaded/com/google/common/base/Preconditions
	at com.google.cloud.bigtable.hbase.async.TestAsyncCheckAndMutate.getBuilder(TestAsyncCheckAndMutate.java:99)
	at com.google.cloud.bigtable.hbase.async.TestAsyncCheckAndMutate.checkAndDelete(TestAsyncCheckAndMutate.java:81)

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java (hunk 3 above)
Comment (pawan-qlogic) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: Implemented by calling existing createTable and sent empty Optional
  /*
   * (non-Javadoc)
   * @see org.apache.hadoop.hbase.client.AsyncAdmin#createTable(org.apache.hadoop.hbase.client.TableDescriptor)
   */
  @Override
  public CompletableFuture<Void> createTable(TableDescriptor desc) {
    return createTable(desc, Optional.empty());
  }

below method's signature is modified, so changed the scope to private 
private CompletableFuture<Void> createTable(TableDescriptor desc, Optional<byte[][]> splitKeys) 

and new signature method is below
  @Override
  public CompletableFuture<Void> createTable(TableDescriptor desc, byte[][] splitKeys) {
    return createTable(desc, Optional.of(splitKeys));
  }

File: pom.xml (hunk 1 above)
Comment (pawan-qlogic) [reply pom.xml]: Please ignore NullPointerException, it is fixed.

**Pull request merged**